### DISCUSSION TITLE
apisix-java-plugin-runner run alone or bundled with apisix, discussing in the prod environment

### DISCUSSION THREAD

**Discussion by arabot777** [apache/apisix#5639] [Q&A] [answered]
Ready by [document](https://github.com/apache/apisix-java-plugin-runner/blob/main/docs/en/latest/how-it-works.md)，I understand that java plugin run in alone is to debug code,  if I want use it in prod env， it best to bundled with apisix which  just to ensure the high availability of Java services ?
In actual use， if I want to reload apisix, it is also reload java service too, and during this period, apisix is not available. 
Two solutions：
1. When I reload apisix, pick the apisix from OSPF. But this is not very friendly
2. run java service in alone, I will guarantee that java service is highly available.  But you don't seem to recommend this way.

Can you give some advice? thanks!

**Comment by tzssangglass** [comment, marked as answer]
> I will guarantee that java service is highly available

This is unnecessary. Because the java plugin runner is only called by APISIX, the lifecycle of apisix-java-plugin-runner is managed by APISIX. We can not run java service in alone.
